shim/build.rs

```diff
@@ -1,6 +1,58 @@
+use std::{env, ffi::OsStr, process::Command, path::PathBuf};
+
+
 fn main() {
-    println!("cargo:rustc-link-search=../cli");
+    let is_release = match env::var("PROFILE") {
+        Ok(profile) if profile == "release" => true,
+        _ => false
+    };
+    let lib_search_path = if is_release {
+        expect_release_lib()
+    } else {
+        build_debug_libturbo()
+    };
+    println!("cargo:rustc-link-search={}", lib_search_path);
     println!("cargo:rustc-link-lib=turbo");
-    println!("cargo:rustc-link-lib=framework=cocoa");
-    println!("cargo:rustc-link-lib=framework=security");
+    if cfg!(target_os = "macos") {
+      println!("cargo:rustc-link-lib=framework=cocoa");
+      println!("cargo:rustc-link-lib=framework=security");
+    }
+}
+
+fn expect_release_lib() -> String {
+    let target = build_target::target().unwrap();
+    let platform = match target.os {
+        build_target::Os::MacOs => "darwin",
+        build_target::Os::Windows => "windows",
+        build_target::Os::Linux => "linux",
+        _ => panic!("unsupported target {}", target.triple)
+    };
+    let arch = match target.arch {
+        build_target::Arch::AARCH64 => "arm64",
+        build_target::Arch::X86_64 => "amd64_v1",
+        _ => panic!("unsupported target {}", target.triple)
+    };
+    let mut dir = PathBuf::from("libturbo");
+    // format is ${BUILD_ID}_${OS}_${ARCH}. Build id is, for goreleaser reasons, turbo-${OS}
+    dir.push(format!("turbo-{}_{}_{}", platform, platform, arch));
```

Comment:
Just noting, you could do:

```suggestion
    dir.push(format!("turbo-{platform}_{platform}_{arch}"));
```

Might be more self documenting